stan/math/prim/mat/fun/matrix_exp_pade.hpp

```diff
@@ -0,0 +1,250 @@
+// This file is part of Eigen, a lightweight C++ template library
+// for linear algebra.
+//
+// Copyright (C) 2009, 2010, 2013 Jitse Niesen <jitse@maths.leeds.ac.uk>
+// Copyright (C) 2011, 2013 Chen-Pang He <jdh8@ms63.hinet.net>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// This file was edited for to the Stan math library to create
+// the matrix exponential function (matrix_exp), 2016.
+
+#ifndef STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_PADE_HPP
+#define STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_PADE_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+
+namespace stan {
```

Comment:
It is from the beta version of Eigen 3.3 in unsupported (so not the currently implemented Eigen version in Stan), and has been modified to be auto-differentialble. Their functions were overloaded for float, double, and long double, but not for var and fvar. I created a template version.